corehq/apps/userreports/reports/builder/forms.py

```diff
@@ -846,14 +869,96 @@ def _get_user_name_pseudo_property():
             data_types=["string"],
         )
 
+
+class ApplicationCaseDataSourceHelper(CaseDataSourceHelper):
+    def __init__(self, domain, app, source_type, source_id):
+        self.app = app
+        assert source_type == 'case'
+        super().__init__(domain, source_type, source_id)
+        prop_map = get_case_properties(
+            self.app, [self.source_id], defaults=list(DEFAULT_CASE_PROPERTY_DATATYPES),
+            include_parent_properties=True,
+        )
+        self.case_properties = sorted(set(prop_map[self.source_id]) | {'closed', 'closed_on'})
+
+    def _get_data_source_build_information(self):
+        return DataSourceBuildInformation(
+            source_id=self.source_id,
+            app_id=self.app._id,
+            app_version=self.app.version,
+        )
+
+    @property
+    def uses_registry_data_source(self):
+        return False
+
     @property
     @memoized
     def data_source_name(self):
         today = datetime.datetime.utcnow().strftime("%Y-%m-%d %H:%M:%S")
         return "{} (v{}) {}".format(self.source_id, self.app.version, today)
 
 
-def get_data_source_interface(domain, app, source_type, source_id):
+class RegistryCaseDataSourceHelper(CaseDataSourceHelper):
+    def __init__(self, domain, registry_slug, source_type, source_id):
+        assert source_type == 'case'
+        self.registry_slug = registry_slug
+        super().__init__(domain, source_type, source_id)
+
+        registry_helper = DataRegistryHelper(self.domain, registry_slug=self.registry_slug)
+        owning_domain = registry_helper.registry.domain
+        prop_map = get_data_dict_props_by_case_type(owning_domain)
+        self.case_properties = sorted(
+            set(prop_map[self.source_id]) | {'closed', 'closed_on'}
+        )
+
+    def _get_data_source_build_information(self):
+        return DataSourceBuildInformation(
+            source_id=self.source_id,
+            registry_slug=self.registry_slug,
+        )
+
+    @property
+    def uses_registry_data_source(self):
+        return True
+
+    @property
+    @memoized
+    def data_source_name(self):
+        today = datetime.datetime.utcnow().strftime("%Y-%m-%d %H:%M:%S")
+        return "{} {} {}".format(self.source_id, self.registry_slug, today)
+
+    @property
+    def data_source_properties(self):
+        properties = super().data_source_properties
+        properties[COMMCARE_PROJECT] = DataSourceProperty(
+            type=PROPERTY_TYPE_RAW,
+            id=COMMCARE_PROJECT,
+            text=_('CommCare Project'),
+            source='domain',
```

Comment:
I think this should be `commcare_project` shouldn't it? (Based on https://github.com/dimagi/commcare-hq/pull/30285/files#diff-eadcb42c8517e2f406bc980c0a5a354943556bbbb6602794faf2f33349fb5ae3R670)